test/unit/middleware/generator-middleware.test.js

```diff
@@ -187,4 +185,37 @@ describe('Generator Middleware', function () {
     await repo.history.wait()
   })
 
+  it('can push new actions during history.wait() while processing a generator', async function () {
```

Comment:
seems like a bit of an obscure test, would things blow up if some thing were to go awry in microcosm?